### PR TITLE
Compatibility for filename without quotes in Content-Disposition header

### PR DESCRIPTION
**Pull request by meg1502**
Fixes:OX-6612

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/kernel/xml/Outcall.java [hunk 1]
```diff
@@ -71,7 +71,7 @@
     private static final String CONTENT_TYPE_FORM_URLENCODED = "application/x-www-form-urlencoded; charset=utf-8";
     private static final Pattern CHARSET_PATTERN = Pattern.compile("(?i)\\bcharset=\\s*\"?([^\\s;\"]*)");
     private static final Pattern CONTENT_DISPOSITION_FILENAME_PATTERN =
-            Pattern.compile("(attachment|inline);\\s*filename\\s*=\\s*\"([^\"]*)\"");
+            Pattern.compile("(attachment|inline);\\s*filename\\s*=\\s*\"?([^\"]*)\"?");
```
Comment (qw3ry) [inline src/main/java/sirius/kernel/xml/Outcall.java:74]: Sicher, dass das so funktioniert?
- Der Filename kann damit das " auch nur vorne ODER hinten haben
- Wenn der Filename ohne " endet, geht der Match trotzdem bis zum nächsten " weiter. Also `filename = test.pdf, charset = "utf8"` würde `test.pdf, charset =` matchen

Comment (qw3ry) [reply src/main/java/sirius/kernel/xml/Outcall.java:74]: Eventuell besser `\"([^\"]*)\"|\w+`, das _sollte_ einige Probleme vermeiden

**Review by sabieber — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

File: src/main/java/sirius/kernel/xml/Outcall.java [hunk 2]
```diff
@@ -71,7 +71,7 @@
     private static final String CONTENT_TYPE_FORM_URLENCODED = "application/x-www-form-urlencoded; charset=utf-8";
     private static final Pattern CHARSET_PATTERN = Pattern.compile("(?i)\\bcharset=\\s*\"?([^\\s;\"]*)");
     private static final Pattern CONTENT_DISPOSITION_FILENAME_PATTERN =
-            Pattern.compile("(attachment|inline);\\s*filename\\s*=\\s*\"([^\"]*)\"");
+            Pattern.compile("(attachment|inline|form-data);.*filename\\s*=\\s*(?<unquoted>\"(?<quoted>[^\"]*)\"|[\\w.-]+)");
```
Comment (idlira) [inline src/main/java/sirius/kernel/xml/Outcall.java:74]: Is it wise to read it from a config setting?
It looks to me that very soon we will find yet another website which delivers other stuff.

Comment (meg1502) [reply src/main/java/sirius/kernel/xml/Outcall.java:74]: this should be a catch all approach, if we find stuff this doesn't match we need to add it to the test and expand the pattern

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/kernel/xml/Outcall.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/kernel/xml/Outcall.java:74]: A config value can also only changed via a patch and we probably do not want this to differ from product to product

**Review by andyHa — APPROVED** [review]

**Pull request merged**